bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/codecs/DICodecsManager.java

```diff
@@ -66,52 +60,60 @@ public DICodecsManager(final Map<String, Codec> codecs) {
    public String getPreferredFormat() { return preferredFormat; }
 
    @Override
-   public Map<String, JsonNode> encode(final EObject eObject) throws EncodingException {
+   public Map<String, JsonNode> encode(final String modelUri, final EObject eObject) throws EncodingException {
       Map<String, JsonNode> encodings = new LinkedHashMap<>();
-      formatToCodec.forEach((key, codec) -> {
-         try {
-            encodings.put(key, codec.encode(eObject));
-         } catch (EncodingException e) {
-            e.printStackTrace();
-         }
+      this.codecProviders.forEach(cp -> {
+         cp.getAllFormats().forEach(f -> {
```

Comment:
Sorry, I missed this in my previous review. We're not considering format priorities, here. We're iterating all available formats from all known providers, but we don't know that we're getting the most appropriate implementation of each format because we can have multiple providers offering the same format.

I think we need to first collect the unique formats supported by all providers, then for each, hold an auction with all providers to get the best codec for the format.